src/more_ops.rs

```diff
@@ -217,9 +170,13 @@ pub fn op_unknown<A: Allocator>(
 
     assert!(cost > 0);
 
+    check_cost(allocator, cost, max_cost)?;
     cost *= cost_multiplier + 1;
-
-    Ok(Reduction(cost as Cost, allocator.null()))
+    if cost > u32::MAX as u64 {
```

Comment:
this check was missing previously, it was an oversight and exposed by one of the new test cases (as a discrepancy with the python implementation)